lms/djangoapps/teams/views.py

```diff
@@ -326,37 +342,49 @@ def get(self, request):
             result_filter.update({'topic_id': request.QUERY_PARAMS['topic_id']})
         if 'include_inactive' in request.QUERY_PARAMS and request.QUERY_PARAMS['include_inactive'].lower() == 'true':
             del result_filter['is_active']
-        if 'text_search' in request.QUERY_PARAMS:
-            return Response(
-                build_api_error(ugettext_noop("text_search is not yet supported.")),
-                status=status.HTTP_400_BAD_REQUEST
-            )
 
         queryset = CourseTeam.objects.filter(**result_filter)
 
-        order_by_input = request.QUERY_PARAMS.get('order_by', 'name')
-        if order_by_input == 'name':
-            queryset = queryset.extra(select={'lower_name': "lower(name)"})
-            queryset = queryset.order_by('lower_name')
-        elif order_by_input == 'open_slots':
-            queryset = queryset.annotate(team_size=Count('users'))
-            queryset = queryset.order_by('team_size', '-last_activity_at')
-        elif order_by_input == 'last_activity_at':
-            queryset = queryset.annotate(team_size=Count('users'))
-            queryset = queryset.order_by('-last_activity_at', 'team_size')
+        has_text_search = 'text_search' in request.QUERY_PARAMS
+        if has_text_search and CourseTeamIndexer.search_is_enabled():
+            search_engine = CourseTeamIndexer.engine()
+            text_search = request.QUERY_PARAMS['text_search'].encode('utf-8')
+
+            search_result = search_engine.search_string(text_search, size=MAXIMUM_SEARCH_SIZE)
+
+            results = search_result.get('results', [])
+            search_queryset_pks = [item['data']['pk'] for item in results]
+            queryset = queryset.filter(pk__in=search_queryset_pks)
         else:
-            return Response({
-                'developer_message': "unsupported order_by value {ordering}".format(ordering=order_by_input),
-                # Translators: 'ordering' is a string describing a way
-                # of ordering a list. For example, {ordering} may be
-                # 'name', indicating that the user wants to sort the
-                # list by lower case name.
-                'user_message': _(u"The ordering {ordering} is not supported").format(ordering=order_by_input),
-            }, status=status.HTTP_400_BAD_REQUEST)
+            # If we are in this block only because search is disabled log a message
+            if has_text_search:
+                log.info(u"Text search requested but disabled")
```

Comment:
That seems fine. We should say though in the API documentation that text_search will be ignored unless the ENABLE_TEAMS_SEARCH feature flag is enabled.